src/plugins/data/common/index_patterns/index_patterns/ensure_default_index_pattern.ts

```diff
@@ -48,23 +47,43 @@ export const createEnsureDefaultIndexPattern = (
       return;
     }
     const patterns = await this.getIds();
-    let defaultId = await uiSettings.get('defaultIndex');
-    let defined = !!defaultId;
+    const defaultId = await uiSettings.get('defaultIndex');
+    const defined = !!defaultId;
     const exists = includes(patterns, defaultId);
-
+    const availablePatterns: string[] = [];
     if (defined && !exists) {
-      await uiSettings.remove('defaultIndex');
-      defaultId = defined = false;
```

Comment:
It seems my fault that asked you to delete this logic because I thought the logic is newly introduced. If it is an existing behavior, please keep it.